src/main/scala/com/gyoho/adget/AdgetService.scala

```diff
@@ -0,0 +1,30 @@
+package com.gyoho.adget
+
+import com.gyoho.adget.utils.ScalaHttpClient
+
+object AdgetService {
+  @volatile private var ads: Map[String, Ad] = Map.empty
```

Comment:
This is possible primary because we construct a new map instead of **mutating the old one's fields** for every update.